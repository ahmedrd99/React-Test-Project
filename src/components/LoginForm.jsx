import { useState } from 'react';

export function LoginForm() {
    // 1. Déclarez le state avec useState
  const [username, setUsername] = useState('');
    return ( 
       
    <form onSubmit= {(e) => {
        e.preventDefault(); // Empêche la page de se recharger
        const formData = new FormData(e.target);//(e.target) ici cible le form lui-méme 
        // new FormData() : Crée un objet qui récupère automatiquement toutes les données du formulaire.
        const username = formData.get("username")
        const password = formData.get("password")
        
         fetch("http://localhost:5173/api/login", {
            body: {
                username,
                password,
            }, method: "POST" , 
         });

     }}
     > 
        <label htmlFor="username"
         style={{cursor:"Pointer",
            fontSize:"32px"
        }}>Username:</label>
        <br />
        <br />
<input
    id='username'
    value={username}
    onChange={(e) => { setUsername(e.target.value);
    console.log(e.target.value ) ;}} 
    name="username"
    style={{padding:"10px", borderRadius:"6px", boxShadow:"initial",width:"300px"

 }}/>
<br />
<br />
<label htmlFor="Email" 
style={{cursor:"Pointer",
        fontSize:"32px"
        }}>Your Email:
</label>
<br />
<br />
<input id='Email'
 name="Email" 
 type="email" 
 style={{padding:"10px", borderRadius:"6px", boxShadow:"initial", width:"300px"}}/>
<br />
<br />
<label htmlFor="Pasword" style={{cursor:"Pointer",
            fontSize:"32px"
        }} >Password:</label>
        <br />
        <br />
<input
    id='Password' 
    name="password" 
    type="password"
    onChange={(e)=> {console.log("lettre:",e.target.value);}} 
    style={{padding:"10px", borderRadius:"6px", boxShadow:"initial", width:"300px"}}/>
<br />
<br />
<button  
onClick={(e) =>{console.log( "buttoncliquer")} }
style={{display:"block", padding:"10px",width:"100px",   fontWeight: "bold" , fontSize:"20px", cursor:"pointer"}}>
    Login
</button>

    </form>
);
}