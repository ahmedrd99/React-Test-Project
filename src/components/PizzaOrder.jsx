import { useState,useEffect } from "react";

    function  PizzaOrder() {
  
      const [counter, setCounter] = useState(0);
      const [pizza, setPizza] = useState("Margherita");
      const [selectedPizzaType, setSelectedPizzaType] = useState("Margherita");
      const [customPizzaName, setCustomPizzaName] = useState("");
      const [userInput, setUserInput] = useState("");
    
      useEffect(() => {
        document.title = `React Tutorial | Pizza: ${pizza} | Compteur: ${counter}`;
      }, [counter, pizza]);
    
      const handlePizzaChange = (newType) => {
        setSelectedPizzaType(newType);
        if (newType !== "Custom") {
          setPizza(newType);
        } else {
          setPizza(customPizzaName || "Custom");
        }
      };
    
      return (
        <div style={{ padding: '20px', fontFamily: 'Arial' }}>
           
    
          <div style={{ margin: '15px 0' }}>
            <p>Pizza actuelle: <strong>{pizza}</strong></p>
            <select 
              value={selectedPizzaType}
              onChange={(e) => handlePizzaChange(e.target.value)} //
              style={{ padding: '5px' }}
            >
              <option value="Margherita">Margherita</option>
              <option value="4 Fromages">4 Fromages</option>
              <option value="Pepperoni">Pepperoni</option>
              <option value="Custom">Personnalis√©e</option>
            </select>
            
          </div>
    
          {selectedPizzaType === "Custom" && (
            <input
              type="text"
              value={customPizzaName}
              placeholder="Nommez votre pizza"
              onChange={(e) => {
                setCustomPizzaName(e.target.value);
                setPizza(e.target.value);
              }}
              style={{ padding: '5px', marginTop: '10px' }}
            />
          )}
        </div>
      );
 

}