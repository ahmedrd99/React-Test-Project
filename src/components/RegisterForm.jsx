import { useState } from "react"

export function RegisterForm(){
    const [usernamen , setUsername]=useState("");
    const [password , setPassword]=useState("");
    const [displayName , setDisplayName] = useState("");
    return <form>
        <div>
            <label htmlFor="username"> Username:</label>
            <br />
            <br />
        <input 
        id="username"
        value={username}
        onChange={(e)=> { setUsername (e.target.value);

          }} />
        
        </div>
         <br />
        <div>
            <label htmlFor="password" >Password: </label>
            <br />
            <br />
        <input id="password" type="text" />
        
        </div>
         <br />
        <div>
            <label htmlFor="displayName"> Display name: </label> 
            <br />
            <br />
        <input id="displayName" type="text" />
        
        </div>
    </form>
}