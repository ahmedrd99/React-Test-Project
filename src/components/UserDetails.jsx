import   PropTypes from "prop-types";
import { useState } from "react";


export function UserDetails({user,setUsers}) {
    const  [isEditing,setIsEditing]=useState(false);
console.log(user);
console.log(isEditing);

const [username,setUsername]=useState(user.username);
const [email,setEmail]  =useState(user.email);


    return (
   <div>
     <div>
        <b>ID:</b> 
        <span>{user.id}</span>
        <br />
        <b>Username:</b> 
         {isEditing? (
            
            <input 
            name="username"
            id="username"
            type="text"
            value={username}
            onChange={(e)=>{  setUsername( e.target.value);}} 
            />
        ) :( <span>{username}</span>)}
        <br />
        <b>Email: </b> 
        {isEditing? (
            
            <input 
            name="email"
            id="email"
            type="email"
            value={email}
            onChange={(e)=> { setEmail ( e.target.value) }} 
            />
        ) : <span>{email}</span>}
  

    </div>
    <div>
        <button 
    onClick={()=> { 
        setIsEditing((currentState)=> !currentState);
    }} > 
          EDIT
        </button>
        <button onClick={() => alert("User deleted")}>DELETE</button>
        
        {isEditing && (
        <button onClick={()=> {
          setUsers(
            (currentUserState)=> {
              return currentUserState.map(
                (currentUser)=>currentUser.id === user.id? 
                {...currentUser,username,email}:currentUser
              )
        }
          );
          setIsEditing(false); 
        }}  >Save</button>
      )}
    </div>
   </div>
   
    );
}


UserDetails.propTypes ={
    user: PropTypes.shape({
        id:PropTypes.number.isRequired,
        username:PropTypes.string.isRequired,
        email:PropTypes.string.isRequired,
        
    }).isRequired,
    setUsers:PropTypes.func.isRequired, 
}




//version corrigé.
/*
import React, { useState } from "react";

export function UserDetails({ user }) {
  const [isEditing, setIsEditing] = useState(false);
  const [email, setEmail] = useState(user.email); // on stocke l'email

  return (
    <div>
      <div>
        <b>ID:</b> <span>{user.id}</span> <br />
        <b>Username:</b> <span>{user.username}</span> <br />
        <b>Email:</b>
        {isEditing ? (
          // Si on est en mode édition ➔ afficher input
          <input
            type="text"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        ) : (
          // Sinon ➔ afficher normalement l'email
          <span>{email}</span>
        )}
      </div>

      <div>
        <button
          onClick={() => {
            setIsEditing((current) => !current);
          }}
        >
          {isEditing ? "Save" : "Edit"}
        </button>
        <button>Delete</button>
      </div>
    </div>
  );
}


*/