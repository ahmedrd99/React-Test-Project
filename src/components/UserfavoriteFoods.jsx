import { createElement } from "react";
import styles from "./users.module.scss";
export function UserFavoriteFoods() {
    return createElement (
        "section",
         null,  
    
         
         <div>
            <span className={styles.foodsTitle}> favorite Foods:</span>
            <br />
            

            <ul> 
                <li> Suchi </li>
                <li>Pizza</li>
                <li>Tacose</li>
            </ul>
         </div>
        
        

    );
}
