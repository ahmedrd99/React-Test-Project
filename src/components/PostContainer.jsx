import { useContext, useState } from "react";
import {PostContent} from "./PostContent";
import { UserContext } from "../utils/Contexts/UserContext";


export function PostContainer () { 
      const [data, setData] = useState(" hello Ahmed ! ");
     const userContextData = useContext(UserContext) ;

    // console.log(userContextData.setUserData);


    return ( 
        <div>
          
            <PostContent data = {data} /> 
            <div>{userContextData.id}</div>
            <div>{userContextData.name}</div>
            <div>{userContextData.username}</div>
            <div>{userContextData.email}</div>
            <PostContent data={data}/>
        </div>
    );
}