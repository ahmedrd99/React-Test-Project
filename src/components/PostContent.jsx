import {PostContentButtons} from "./PostContentButtons";
import { useContext } from "react"; // Import manquant
import { UserContext } from "../utils/Contexts/UserContext";

export function PostContent ({data}) { 
    const userContextData = useContext(UserContext);
    return ( 
        <div>
             
            <PostContentButtons data={data} /> 
             
        </div>
    );
}