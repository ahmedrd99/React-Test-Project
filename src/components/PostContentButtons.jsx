import { UserContext } from "../utils/Contexts/UserContext";
import { PostContent } from "./PostContent";
import { useContext } from "react"; // Ajout crucial
 

export function PostContentButtons ({data}) { 
  const {id,setUserData} = useContext( UserContext);// On récupère les données
    return ( 
    
        <div>
             
            {id}
            <br />
            <button onClick={()=> {
                setUserData((currentState)=> ({...currentState,
                     name: 'updated Display Name'}))
            }}> Click Me</button>
         
        </div>
    );
}