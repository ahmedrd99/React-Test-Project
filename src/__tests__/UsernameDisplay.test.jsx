import { describe,it,test,expect} from "vitest";
import { UsernameDisplay } from "../components/usernameDisplay";
import { render,screen } from "@testing-library/react";
import { element } from "prop-types";
 
describe(' UsernameDisplay', ()=> { 
it('should rendre username ',async ()=> {
    render(< UsernameDisplay username="ahmeddev"  />);
  // expect( result.container) .toMatchSnapshot()

expect (   await screen.findByText("ahmeddev",{},{ timeout:10000})).toBeInTheDocument(); // Recherche + VÃ©rification
// element.array.forEach((element) =>  expect(element).not.toBeInTheDocument());
  //  let error;
// try {
//  screen.getByText("ahmeddev")
//    }catch(err) {
//     error = err;
//    }
//    expect(error).toBeDefined()
});
}); 