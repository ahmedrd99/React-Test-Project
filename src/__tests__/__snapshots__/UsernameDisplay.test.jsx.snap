// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` UsernameDisplay > should rendre username  1`] = `
<div>
  <div>
    <span>
      ahmeddev
       
    </span>
  </div>
</div>
`;
